scoring_board.cpp

```diff
@@ -0,0 +1,90 @@
+#include "scoring_board.h"
+#include <cassert>
+#include <cmath>
+#include <string>
+
+scoring_board::scoring_board(unsigned int score_player1,
+                             unsigned int score_player2 ,
+                             unsigned int score_player3,
+                             unsigned int timer,
+                             short winner)
+    : m_score_player1{score_player1},
+      m_score_player2{score_player2},
+      m_score_player3{score_player3},
+      m_timer{timer},
+      m_winner{winner}
+{
+
+}
+
+void scoring_board::set_winner(short winner)
+{
+    if ((winner >= -1) & (winner <= 2))
```

Comment:
I agree that doing this might plant hidden danger here. A workaround may be that we create a enum class of winners, give it a method `winner.get_index()` to convert it to index.

Reply:
I like this solution :)

Reply:
Made a commit to fix this using throw :)